File: src/com/basmilius/ps/bastools/util/JUtils.java
package com.basmilius.ps.bastools.util;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.wm.IdeFrame;
import com.intellij.openapi.wm.WindowManager;

import javax.swing.*;

public final class JUtils
{

	/**
	 * Gets the root component for the project.
	 *
	 * @param project Project.
	 *
	 * @return JComponent
	 */
	public static JComponent getRootComponent (final Project project)
	{
		if (project != null)
		{
			final IdeFrame frame = WindowManager.getInstance().getIdeFrame(project);

			if (frame != null)
			{
				return frame.getComponent();
			}
		}

		final JFrame frame = WindowManager.getInstance().findVisibleFrame();
		return frame != null ? frame.getRootPane() : null;
	}

}

File: src/com/basmilius/ps/bastools/util/JUtils.kt
package com.basmilius.ps.bastools.util

import com.intellij.openapi.project.Project
import com.intellij.openapi.wm.WindowManager

import javax.swing.*

object JUtils
{

	/**
	 * Gets the root component for the project.
	 *
	 * @param project Project.
	 *
	 * @return JComponent
	 */
	fun getRootComponent(project: Project?): JComponent?
	{
		if (project != null)
		{
			val frame = WindowManager.getInstance().getIdeFrame(project)

			if (frame != null)
			{
				return frame.component
			}
		}

		val frame = WindowManager.getInstance().findVisibleFrame()
		return frame?.rootPane
	}

}
